### ISSUE TITLE
Setting Tabby as default terminal program?

### ISSUE THREAD

**Issue by DutchEclipse** [Eugeny/tabby#4882]
Is it possible to set Tabby as the default terminal program in Windows 10 and Windows 11?
![image](https://user-images.githubusercontent.com/83969413/140619138-48fe2abd-e0da-4601-941e-b2dc8dc6a588.png)

**Comment by Eugeny** [comment]
AFAIK Windows doesn't have a concept of a default terminal. On the screenshot is the default profile selector in the Windows Terminal.

**Comment by GameBurrow** [comment]
That selection in Win 11 and newer versions of Win 10 actually do set the console application, all regular scripts (non-admin) do open in the selected terminal. It's a newer feature they added last year.

Same setting can found in powershell and cmd settings:
![image](https://user-images.githubusercontent.com/4341091/151718442-0db52294-446b-474f-b228-6779cc3ed47d.png)

**Comment by Comnenus** [comment]
I would like to be able to do this, also.

**Comment by ysc3839** [comment]
Related PR: https://github.com/microsoft/terminal/pull/7489

**Comment by ysc3839** [comment]
This feature need this PR to be merged: microsoft/node-pty#716

**Comment by ysc3839** [comment]
@Eugeny Hello, I'm working on this feature now. I'm not familiar with tabby plugin development, so I need your help.
The workflow is:
1. The plugin registers terminal handoff callback.
2. When an external console app starts, the callback will be called with some handles.
3. The handles need to be passed to `node-pty` and get a useable pty instance.
4. Create a new tab with that pty instance.

The modified `node-pty` is at https://github.com/ysc3839/node-pty/commit/1feb412e4527ff022ccbf2b68d9fa9ff4e3e84d8
The module to register callback is still working in progress.
Basically the usage is:
```js
const pty = require('node-pty');
const handoff = require('terminal-handoff');

handoff.register(/*clsid=*/'{E12CFF52-A866-4C77-9A90-F570A7AA2C6B}', /*callback=*/function (input, output, signal, ref, server, client) {
  const handoffHandles = { input, output, signal, ref, server, client };
  const pty = pty.handoff(handoffHandles);
}, /*once=*/false);
```

A demo video:

https://github.com/user-attachments/assets/c7214fff-a01d-441d-995f-1f43e828f810

**Comment by ysc3839** [comment]
The module to register callback is https://github.com/ysc3839/node-terminal-handoff

**Comment by Lisoveliy** [comment]
So, what's in there? @ysc3839 Do you have any progress in this?

**Comment by elysious** [comment]
anything new..?

**Comment by Mikachu2333** [comment]
Any resolution?
